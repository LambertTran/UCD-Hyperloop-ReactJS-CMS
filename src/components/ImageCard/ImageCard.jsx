import React from 'react';
import { Link } from 'react-router-dom';

export default class ImageCard extends React.Component {

  render() {
    return (
      <div className="image-card">
        <Link to={this.props.subteam}>
          <h3>{this.props.subteam}</h3> 
          <img src={this.props.img} alt=""/>
        </Link>   
      </div>
    )
  }
}