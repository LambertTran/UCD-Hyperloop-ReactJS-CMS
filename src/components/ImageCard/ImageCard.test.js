// packages
import React from 'react';
import { shallow } from 'enzyme';
import setupTests from '../../shares/setupTests';

// components
import ImageCard from './ImageCard';

describe('ImageCard', () => {
  // initiate sample props
  const img = 'http://kb4images.com//images/image/37272651-image.jpg';
  const subteam = 'business-team';
  const props = {img,subteam};

  // mock component
  const mockImageCard = jest.fn();
  const imageCard = shallow( <ImageCard {...props} /> );

  it('renders correctly', () => {
    expect(imageCard).toMatchSnapshot();
  })

  describe('when props are passed into this component', () => {{
    it('has image link', () => {
      expect(imageCard.find('.image-card img').prop('src')).toEqual(img);
    })
    
    it('has correct url and title', () => {
      expect(imageCard.find('.image-card h3').text()).toEqual(subteam);
    })
  }})

  describe('when clicking on `ImageCard`', () => {

    beforeEach(() =>{
      imageCard.find('.image-card').simulate('click');
    })
  })
})