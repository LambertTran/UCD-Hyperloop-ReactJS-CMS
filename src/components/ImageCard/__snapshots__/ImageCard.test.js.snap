// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageCard
    img="http://kb4images.com//images/image/37272651-image.jpg"
    subteam="business-team"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="business-team"
>
        <h3>
                business-team
        </h3>
        <img
                alt=""
                src="http://kb4images.com//images/image/37272651-image.jpg"
        />
</Link>,
      "className": "image-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h3>
            business-team
</h3>,
          <img
            alt=""
            src="http://kb4images.com//images/image/37272651-image.jpg"
/>,
        ],
        "replace": false,
        "to": "business-team",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "business-team",
          },
          "ref": null,
          "rendered": "business-team",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "http://kb4images.com//images/image/37272651-image.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="business-team"
>
          <h3>
                    business-team
          </h3>
          <img
                    alt=""
                    src="http://kb4images.com//images/image/37272651-image.jpg"
          />
</Link>,
        "className": "image-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h3>
              business-team
</h3>,
            <img
              alt=""
              src="http://kb4images.com//images/image/37272651-image.jpg"
/>,
          ],
          "replace": false,
          "to": "business-team",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "business-team",
            },
            "ref": null,
            "rendered": "business-team",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "http://kb4images.com//images/image/37272651-image.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
